src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py

```diff
@@ -144,3 +144,11 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs):  # pyli
 register_cli_argument('appservice web source-control', 'branch', help='the branch name of the repository')
 register_cli_argument('appservice web source-control', 'repository_type', help='repository type', default='git', **enum_choice_list(['git', 'mercurial']))
 register_cli_argument('appservice web source-control', 'git_token', help='git access token required for auto sync')
+
+register_cli_argument('functionapp', 'name', configured_default='web',
+                      arg_type=name_arg_type, completer=get_resource_name_completion_list('Microsoft.Web/sites'), id_part='name',
```

Comment:
This completer is going to yield incorrect values. It's going to return all sites, not just function apps.

In fact, this logic bug will apply to all commands that are split out between function apps and webapps. Since they are all just webapps, you are going to be able to delete function apps with `webapp delete` and delete webapps with `functionapp delete`.  You will get identical output for `functionapp list` and `webapp list`.